http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java

```diff
@@ -0,0 +1,184 @@
+
+import static io.quarkus.ts.http.advanced.reactive.MediaTypeResource.APPLICATION_YAML;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.PartType;
+import org.jboss.resteasy.reactive.RestForm;
+import org.yaml.snakeyaml.Yaml;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+import io.quarkus.logging.Log;
+
+@Path("/city")
+public class CityResource {
+
+    private static final String YAML_FILE_PATH = "payload.yaml";
+
+    private static final Logger LOG = Logger.getLogger(CityResource.class);
+
+    private final ObjectMapper objectMapper = new YAMLMapper();
+
+    @Inject
+    private CityListWrapperSerializer serializer;
+
+    private List<City> cityList = new ArrayList<>();
+
+    @GET
+    @Produces(value = MediaType.APPLICATION_XML)
+    public CityListWrapper getCities() {
+        LOG.info("Received request to getCities");
+
+        if (cityList.isEmpty()) {
+            cityList.add(new City("San Bernardino", "EEUU"));
+            cityList.add(new City("Brno", "Czech Republic"));
+            cityList.add(new City("Zaragoza", "Spain"));
+        }
+        return new CityListWrapper(cityList);
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_XML)
+    @Produces(value = MediaType.APPLICATION_XML)
+    public String createCity(String xml) {
+        LOG.info("Received XML payload: " + xml);
+
+        CityListDTO cityListDTO = serializer.fromXML(xml);
+
+        if (cityListDTO == null) {
+            LOG.error("Error deserializing XML");
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("<error>Invalid XML payload</error>")
+                    .build().toString();
+        }
+
+        List<City> cities = cityListDTO.getCityList();
+
+        for (City city : cities) {
+            LOG.info("Saving city " + city);
+        }
+
+        String responseXML = serializer.toXML(cityListDTO);
+
+        return responseXML;
+    }
+
+    @POST
+    @Path("/cities")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes("application/x-yaml")
+    public Object handleConsumedYamlPostRequest(@RequestBody final String yamlPayload) throws IOException {
+
+        LOG.info("Received YAML payload: " + yamlPayload);
+
+        try {
+            return objectMapper.readValue(yamlPayload, Object.class);
```

Comment:
This looks like a smoke test for `YAMLMapper` as deserializing it as object is lenient. While I think it's nice to know that the `YAMLMapper` works in native, I don't think this is how users should write it. 

Microservice won't have one endpoint, right? It depends on use case, but more often than not will endpoints consume same content type (at least we can say more than one)? If your answer is yes, than having boilerplate in every endpoint method doesn't make sense to me. That is why I think users should consume DTOs unless in special cases. I think you test transfer of string and if you didn't deserialize it with YamlMapper, it could as well be JSON etc. and the test would pass with same consumed content type.

In short: there is not a single endpoint that consumes DTO and I disagree with it.

Reply:
Well, in fact with my last changes, now there is an endpoint that consumes DTO https://github.com/quarkus-qe/quarkus-test-suite/blob/17584497b1ca70f5a5382d8c3dd363ab9f7fafcf/http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java#L65

Reply:
oh srry , this  @Consumes(MediaType.APPLICATION_XML)..., ok I will try to think it around to change it.

Reply:
With the last changes I think this thread we could mark it as resolved, wdyt @michalvavrik ?